app/controllers/users.js

```diff
@@ -0,0 +1,43 @@
+const logger = require('../logger'),
+  bcrypt = require('bcryptjs'),
+  User = require('../models').users,
+  errors = require('../errors');
+
+exports.create = (request, response, next) => {
+  const saltRounds = 10;
+  const regexWoloxMail = /[\w\d._]+@wolox[\w.]+/;
+  const regexPassword = /[\w\d]{8,}/;
+
+  const { firstName, lastName, email, password } = request.body;
+
+  const validationErrors = [];
+
+  const pushRequired = field => validationErrors.push(`The ${field} is required`);
+
+  if (!firstName) pushRequired('firstName');
+  if (!lastName) pushRequired('lastName');
+
+  if (!email) pushRequired('email');
+  else if (!regexWoloxMail.test(email)) validationErrors.push(`The email has an invalid format`);
+
+  if (!password) pushRequired('password');
+  else if (!regexPassword.test(password))
+    validationErrors.push(`The password is invalid. It must be alphanumeric and a minimum of 8 characters`);
+
+  if (validationErrors.length > 0)
```

Comment:
You did a really nice job with all this validations! But i think it would be better to have them outside the controller methods. For that,  we will need to use middlewares; we will talk later about how to do this